crates/wasmparser/src/validator/core/canonical.rs

```diff
@@ -140,18 +140,18 @@ impl<'a> TypeCanonicalizer<'a> {
         self
     }
 
-    /// Configure canonicalization to transform all type indices into
-    /// `CoreTypeId`s, even from within the same rec group.
-    pub fn with_only_ids(&mut self) -> &mut Self {
-        self.mode = CanonicalizationMode::OnlyIds;
-        self
-    }
+    // /// Configure canonicalization to transform all type indices into
+    // /// `CoreTypeId`s, even from within the same rec group.
+    // pub fn with_only_ids(&mut self) -> &mut Self {
+    //     self.mode = CanonicalizationMode::OnlyIds;
+    //     self
+    // }
 
-    pub fn within_rec_group(&mut self, types: &TypeList, rec_group_id: RecGroupId) -> &mut Self {
-        assert_eq!(self.mode, CanonicalizationMode::OnlyIds);
-        self.within_rec_group = Some(types[rec_group_id].clone());
-        self
-    }
+    // pub fn within_rec_group(&mut self, types: &TypeList, rec_group_id: RecGroupId) -> &mut Self {
+    //     assert_eq!(self.mode, CanonicalizationMode::OnlyIds);
+    //     self.within_rec_group = Some(types[rec_group_id].clone());
+    //     self
+    // }
```

Comment:
A little surprised to see this stuff commented out. Haven't dug into the `types.rs` changes yet, but I'd expect that we would still want to canonicalize all type references to `CoreTypeId`s before exposing them to users. Like, I don't think they really know what to do with a rec-group local index.